lib/Service/InstallerService.php

```diff
@@ -83,7 +83,7 @@ public function installForceUpdate(ExtensionConfig $config = null): void
         } catch (Exception $couldNotInstall) {
```

Comment:
note ; wouldn't it be better to specailly catch a composer installation / update exception rather than `Exception` ?

Reply:
This "layer" is agnostic to composer

Reply:
So maybe instead, you could in the composer installer (the one in the adapter) catch a composer exception and rethrow this exception in particular ?

Catching `\Exception` (or `\Throwable` ?), unless you _really_ want a catch all (usually, these are not in services, but in front controllers nay ?) should really be limited IMO.

Reply:
Looking again, I think this is also to do with rolling back the configuration.

If something went wrong, whatever it was, I assume the changes are high that the extension was not installed, so we rollback the configuration.

But in general I agree that if that's not a concern, then it would be better to make the exception part of the contract.